internal/ibmcloud-janitor/resources/vpc_lbs.go

```diff
@@ -0,0 +1,116 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resources
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/IBM/vpc-go-sdk/vpcv1"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+type VPCLoadBalancer struct{}
+
+var lbTimeout = time.Minute * 10
+var lbWaitTime = time.Minute * 1
+
+// Cleans up the load balancers in a given region
+func (VPCLoadBalancer) cleanup(options *CleanupOptions) error {
+	resourceLogger := logrus.WithFields(logrus.Fields{"resource": options.Resource.Name})
+	resourceLogger.Info("Cleaning up the load balancers")
+	client, err := NewVPCClient(options)
+	if err != nil {
+		return errors.Wrap(err, "couldn't create VPC client")
+	}
+
+	var lbList map[string]bool
+
+	f := func(start string) (bool, string, error) {
+		listLbOpts := &vpcv1.ListLoadBalancersOptions{}
+		if start != "" {
+			listLbOpts.Start = &start
+		}
+
+		loadBalancers, _, err := client.ListLoadBalancers(listLbOpts)
+		if err != nil {
+			return false, "", errors.Wrap(err, "failed to list the load balancers")
+		}
+
+		if loadBalancers == nil || len(loadBalancers.LoadBalancers) <= 0 {
+			resourceLogger.Info("there are no available load balancers to delete")
+			return true, "", nil
+		}
+
+		for _, lb := range loadBalancers.LoadBalancers {
+			if *lb.ResourceGroup.ID == client.ResourceGroupID {
+				if _, err := client.DeleteLoadBalancer(&vpcv1.DeleteLoadBalancerOptions{
+					ID: lb.ID,
+				}); err != nil {
+					return false, "", errors.Wrapf(err, "failed to delete the load balancer %q", *lb.Name)
+				}
+			}
+			lbList[*lb.ID] = false
+			resourceLogger.WithField("name", *lb.Name).Info("load balancer deleted")
+		}
+
+		if loadBalancers.Next != nil && *loadBalancers.Next.Href != "" {
+			return false, *loadBalancers.Next.Href, nil
+		}
+
+		return true, "", nil
+	}
+
+	if err = pagingHelper(f); err != nil {
+		return errors.Wrapf(err, "failed to delete the load balancer")
+	}
+
+	// check if the LBs are properly deleted
+	if err := checkLBs(lbList, client); err != nil {
+		return errors.Wrapf(err, "failed to check the deletion of load balancers")
+	}
+
+	resourceLogger.Info("Successfully deleted the load balancers")
+	return nil
+}
+
+func checkLBs(list map[string]bool, client *IBMVPCClient) error {
+	timeout := 0 * time.Minute
+	for timeout < lbTimeout {
+		for id, lbDeleted := range list {
+			lbID := id
+			if !lbDeleted {
+				lb, _, err := client.GetLoadBalancer(&vpcv1.GetLoadBalancerOptions{ID: &lbID})
+				if err != nil && strings.Contains(err.Error(), "cannot be found") {
+					list[id] = true
+					continue
+				} else if *lb.ProvisioningStatus == "delete_pending" {
+					break
+				} else {
+					return err
+				}
+			}
+		}
+		fmt.Println("load balancer deletion is pending, checking again in a minute")
```

Comment:
Is it ok to use fmt.Println instead of resourceLogger.Info here?

Reply:
Changed it.